js/share_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2020 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package js
+
+import (
+	"context"
+	"io/ioutil"
+	"testing"
+
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/lib/testutils"
+	"github.com/loadimpact/k6/stats"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitContextNewShareable(t *testing.T) {
+	data := `'use strict';
+function generateArray() {
+    console.log("once");
+    var n = 50;
+    var arr = new Array(n);
+    for (var i = 0 ; i <n; i++) {
+        arr[i] = "something" +i;
```

Comment:
Would the current implementation be usable for binary data, via `ArrayBuffer`, `TypedArray`, etc.?

While a plain array is definitely useful, I think a big use case will be sharing binary data as well. If this is currently possible, a test that showcases it would be helpful.

Reply:
Do you mean like the top array to be ArrayBuffer? I guess it will ... kind of work ... maybe, but will be pretty bad IMO. We should just try to get [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) at some point.

If you ask for if it can be inside the array ... I guess - no, as I have to jsonify the data and then unjsonify it ...

Given how unusable ArrayBuffer is currently in k6 - I would argue this will be fixed when we work on it if it;s possible

Reply:
Right, it would make sense to support this eventually once #1020 is fixed. And even `SharedArrayBuffer`, I wasn't aware of it 👍